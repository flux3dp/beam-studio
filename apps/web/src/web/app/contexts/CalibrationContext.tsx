import React, { createContext, useState } from 'react';

import DeviceMaster from 'helpers/device-master';
import PreviewModeController from 'app/actions/beambox/preview-mode-controller';
import storage from 'implementations/storage';
import versionChecker from 'helpers/version-checker';
import { CameraConfig } from 'interfaces/Camera';
import {
  DEFAULT_CAMERA_OFFSET,
  STEP_ASK_READJUST,
  STEP_PUT_PAPER,
} from 'app/constants/camera-calibration-constants';
import { IDeviceInfo } from 'interfaces/IDevice';

interface CalibrationContextType {
  borderless: boolean;
  cameraPosition: { x: number; y: number };
  setCameraPosition: (val: { x: number; y: number }) => void;
  calibratedMachines: string[];
  setCalibratedMachines: (val: string[]) => void;
  currentOffset: CameraConfig;
  setCurrentOffset: (val: CameraConfig) => void;
  currentStep: symbol;
  setCurrentStep: (val: symbol) => void;
  lastConfig: CameraConfig;
  setLastConfig: (val: CameraConfig) => void;
  gotoNextStep: (step: symbol) => void;
  onClose: (completed: boolean) => void;
  originFanSpeed: number;
  setOriginFanSpeed: (val: number) => void;
  imgBlobUrl: string;
  setImgBlobUrl: (v: string) => void;
  device: IDeviceInfo;
  unit: string;
}

export const CalibrationContext = createContext<CalibrationContextType>({
  borderless: false,
  cameraPosition: { x: 0, y: 0 },
  setCameraPosition: () => {},
  calibratedMachines: [],
  setCalibratedMachines: () => {},
  currentOffset: null,
  setCurrentOffset: () => {},
  currentStep: STEP_PUT_PAPER,
  setCurrentStep: () => {},
  lastConfig: null,
  setLastConfig: () => {},
  gotoNextStep: () => {},
  onClose: () => {},
  imgBlobUrl: '',
  setImgBlobUrl: () => {},
  originFanSpeed: 100,
  setOriginFanSpeed: () => {},
  device: null,
  unit: 'mm',
});

interface CalibrationProviderProps {
  children: React.ReactNode;
  borderless: boolean;
  device: IDeviceInfo;
  onClose: (completed: boolean) => void;
}

let calibratedMachines = [];

export function CalibrationProvider({
  children,
  borderless,
  device,
  onClose,
}: CalibrationProviderProps): JSX.Element {
  const didCalibrate = calibratedMachines.includes(device.uuid);
  const initStep: symbol = didCalibrate ? STEP_ASK_READJUST : STEP_PUT_PAPER;
  const [currentStep, setCurrentStep] = useState(initStep);
  const [currentOffset, setCurrentOffset] = useState(DEFAULT_CAMERA_OFFSET);
  const [lastConfig, setLastConfig] = useState(DEFAULT_CAMERA_OFFSET);
  const setCalibratedMachines = (newList: string[]) => {
    calibratedMachines = newList;
  };
  const [imgBlobUrl, setImgBlobUrl] = useState('');
  const [cameraPosition, setCameraPosition] = useState({ x: 0, y: 0 });
  const [originFanSpeed, setOriginFanSpeed] = useState(1000);
  const unit = (storage.get('default-units') as string) || 'mm';

  const wrappedOnClose = async (completed: boolean) => {
    onClose(completed);
    await PreviewModeController.end({ shouldWaitForEnd: true });
    const tempCmdAvailable = versionChecker(device?.version).meetRequirement('TEMP_I2C_CMD');
    if (originFanSpeed && !tempCmdAvailable) {
      await DeviceMaster.setFan(originFanSpeed);
    }
  };

  return (
    <CalibrationContext.Provider
      value={{
        borderless,
        device,
        currentStep,
        setCurrentStep,
        cameraPosition,
        setCameraPosition,
        currentOffset,
        setCurrentOffset,
        lastConfig,
        setLastConfig,
        calibratedMachines,
        setCalibratedMachines,
        imgBlobUrl,
        setImgBlobUrl,
        originFanSpeed,
        setOriginFanSpeed,
        gotoNextStep: setCurrentStep,
        unit,
        onClose: wrappedOnClose,
      }}
    >
      {children}
    </CalibrationContext.Provider>
  );
}
