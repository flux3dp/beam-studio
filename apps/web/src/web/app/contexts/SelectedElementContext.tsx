import React, { createContext, memo, useCallback, useEffect, useState, useRef } from 'react';

import eventEmitterFactory from 'helpers/eventEmitterFactory';

const canvasEventEmitter = eventEmitterFactory.createEventEmitter('canvas');

export interface SelectedElementContextType {
  selectedElement: Element | null;
}

export const SelectedElementContext = createContext<SelectedElementContextType>({
  selectedElement: null,
});

interface Props {
  children: React.ReactNode;
}

export const SelectedElementContextProvider = memo(({ children }: Props): JSX.Element => {
  const [selectedElement, setSelectedElement] = useState<Element | null>(null);
  const selectedElementRef = useRef<Element | null>(null);

  const handleSetSelectedElem = useCallback(
    (elem: Element): void => {
      if (elem === selectedElementRef.current) return;
      selectedElementRef.current = elem;
      setSelectedElement((cur) => {
        if (cur === elem) return cur;
        (document.activeElement as HTMLInputElement).blur();
        return elem;
      });
    },
    []
  );

  useEffect(() => {
    canvasEventEmitter.on('SET_SELECTED_ELEMENT', handleSetSelectedElem);
    return () => {
      canvasEventEmitter.off('SET_SELECTED_ELEMENT', handleSetSelectedElem);
    };
  }, [handleSetSelectedElem]);

  return (
    <SelectedElementContext.Provider value={{ selectedElement }}>
      {children}
    </SelectedElementContext.Provider>
  );
});

export default {
  SelectedElementContext,
  SelectedElementContextProvider,
};
