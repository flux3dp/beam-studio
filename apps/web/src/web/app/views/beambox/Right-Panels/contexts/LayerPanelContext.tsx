import React, { createContext, useEffect, useState, useCallback, useMemo } from 'react';

import doLayersContainsVector from 'helpers/layer/check-vector';
import eventEmitterFactory from 'helpers/eventEmitterFactory';
import useForceUpdate from 'helpers/use-force-update';
import useWorkarea from 'helpers/hooks/useWorkarea';
import { getLayerElementByName } from 'helpers/layer/layer-helper';
import { promarkModels } from 'app/actions/beambox/constant';

interface ILayerPanelContext {
  selectedLayers: string[];
  setSelectedLayers: (selectedLayers: string[]) => void;
  forceUpdate: () => void;
  forceUpdateSelectedLayers: () => void;
  hasVector: boolean;
  hasGradient: boolean;
}

export const LayerPanelContext = createContext<ILayerPanelContext>({
  selectedLayers: [],
  setSelectedLayers: () => {},
  forceUpdate: () => {},
  forceUpdateSelectedLayers: () => {},
  hasVector: false,
  hasGradient: false,
});
const layerPanelEventEmitter = eventEmitterFactory.createEventEmitter('layer-panel');

interface Props {
  children?: React.ReactNode;
}

export const LayerPanelContextProvider = ({ children }: Props): JSX.Element => {
  const [hasVector, setHasVector] = useState<boolean>(false);
  const [hasGradient, setHasGradient] = useState<boolean>(false);
  const [selectedLayers, setSelectedLayers] = useState<string[]>([]);
  const forceUpdate = useForceUpdate();
  const forceUpdateSelectedLayers = useCallback(
    () => setSelectedLayers([...selectedLayers]),
    [selectedLayers]
  );
  const workarea = useWorkarea();
  const isPromark = useMemo(() => promarkModels.has(workarea), [workarea]);

  const lazySetSelectedLayers = useCallback(
    (newLayers: string[]) => {
      if (
        newLayers.length === selectedLayers.length &&
        newLayers.every((name, i) => name === selectedLayers[i])
      ) {
        return;
      }
      setSelectedLayers(newLayers);
    },
    [selectedLayers, setSelectedLayers]
  );

  useEffect(() => {
    layerPanelEventEmitter.on('UPDATE_LAYER_PANEL', forceUpdate);
    return () => {
      layerPanelEventEmitter.removeListener('UPDATE_LAYER_PANEL', forceUpdate);
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useEffect(() => {
    layerPanelEventEmitter.on('SET_SELECTED_LAYERS', lazySetSelectedLayers);
    return () => {
      layerPanelEventEmitter.removeListener('SET_SELECTED_LAYERS', lazySetSelectedLayers);
    };
  }, [lazySetSelectedLayers]);

  useEffect(() => {
    const getSelectedLayers = (response: { selectedLayers: string[] }) => {
      response.selectedLayers = selectedLayers;
    };
    layerPanelEventEmitter.on('GET_SELECTED_LAYERS', getSelectedLayers);
    const checkVector = () => {
      const newVal = doLayersContainsVector(selectedLayers);
      setHasVector(newVal);
    };
    layerPanelEventEmitter.on('CHECK_VECTOR', checkVector);
    return () => {
      layerPanelEventEmitter.removeListener('GET_SELECTED_LAYERS', getSelectedLayers);
      layerPanelEventEmitter.removeListener('CHECK_VECTOR', checkVector);
    };
  }, [selectedLayers]);

  useEffect(() => {
    const checkGradient = () => {
      if (isPromark) {
        const newVal = selectedLayers.some((layerName: string) => {
          const layer = getLayerElementByName(layerName);
          return !!layer?.querySelector('image[data-shading="true"]');
        });
        setHasGradient(newVal);
      }
    };
    checkGradient();
    layerPanelEventEmitter.on('CHECK_GRADIENT', checkGradient);
    return () => {
      layerPanelEventEmitter.removeListener('CHECK_GRADIENT', checkGradient);
    };
  }, [selectedLayers, isPromark]);

  // move doLayersContainsVector and setVector in effect to avoid heavy render process
  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(() => {
    const newVal = doLayersContainsVector(selectedLayers);
    if (hasVector !== newVal) {
      setHasVector(newVal);
    }
    if (isPromark) {
      const newGradientVal = selectedLayers.some((layerName: string) => {
        const layer = getLayerElementByName(layerName);
        return !!layer?.querySelector('image[data-shading="true"]');
      });
      if (newGradientVal !== hasGradient) setHasGradient(newGradientVal);
    }
  });

  return (
    <LayerPanelContext.Provider
      value={{
        selectedLayers,
        setSelectedLayers: lazySetSelectedLayers,
        forceUpdate,
        forceUpdateSelectedLayers,
        hasVector,
        hasGradient,
      }}
    >
      {children}
    </LayerPanelContext.Provider>
  );
};
