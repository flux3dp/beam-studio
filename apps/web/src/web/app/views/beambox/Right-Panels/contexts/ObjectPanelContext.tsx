import React, { useCallback, useEffect, useRef, useState } from 'react';

import eventEmitterFactory from 'helpers/eventEmitterFactory';
import useForceUpdate from 'helpers/use-force-update';

interface IObjectPanelContext {
  polygonSides: number;
  dimensionValues: any;
  activeKey: string | null;
  updateDimensionValues: (newValues: any) => void;
  getDimensionValues: (
    response: {
      dimensionValues: any;
    },
    key?: string
  ) => any;
  updateActiveKey: (activeKey: string | null) => void;
  updateObjectPanel: () => void;
}

export const ObjectPanelContext = React.createContext<IObjectPanelContext>({
  polygonSides: 5,
  dimensionValues: {},
  activeKey: null,
  updateDimensionValues: () => {},
  getDimensionValues: () => {},
  updateActiveKey: () => {},
  updateObjectPanel: () => {},
});
const objectPanelEventEmitter = eventEmitterFactory.createEventEmitter('object-panel');
const minRenderInterval = 200;

interface Props {
  children?: React.ReactNode;
}

export const ObjectPanelContextProvider = ({ children }: Props): JSX.Element => {
  const forceUpdate = useForceUpdate();
  const [polygonSides, setPolygonSides] = useState(5);
  const [activeKey, setActiveKey] = useState<string | null>(null);
  const dimensionValues = useRef<{ [key: string]: number | string }>({});
  const lastUpdateTime = useRef(Date.now());
  const updateTimeout = useRef<NodeJS.Timeout>(null);
  useEffect(() => {
    objectPanelEventEmitter.on('UPDATE_POLYGON_SIDES', setPolygonSides);
    objectPanelEventEmitter.on('UPDATE_ACTIVE_KEY', setActiveKey);
    return () => {
      objectPanelEventEmitter.removeListener('UPDATE_POLYGON_SIDES', setPolygonSides);
      objectPanelEventEmitter.removeListener('UPDATE_ACTIVE_KEY', setActiveKey);
    }
  }, []);

  const getDimensionValues = useCallback((response: { dimensionValues: any }, key?: string) => {
    response.dimensionValues = key ? dimensionValues.current[key] : dimensionValues.current;
  }, []);

  useEffect(() => {
    objectPanelEventEmitter.on('GET_DIMENSION_VALUES', getDimensionValues);
    return () => {
      objectPanelEventEmitter.removeListener('GET_DIMENSION_VALUES', getDimensionValues);
    }
  }, [getDimensionValues]);
  const updateDimensionValues = useCallback((newValues: { [key: string]: number | string }) => {
    dimensionValues.current = {
      ...dimensionValues.current,
      ...newValues,
    };
  }, []);

  useEffect(() => {
    objectPanelEventEmitter.on('UPDATE_DIMENSION_VALUES', updateDimensionValues);
    return () => {
      objectPanelEventEmitter.removeListener('UPDATE_DIMENSION_VALUES', updateDimensionValues);
    }
  }, [updateDimensionValues]);
  const getActiveKey = useCallback((response: { activeKey: string | null }) => {
    response.activeKey = activeKey;
  }, [activeKey]);

  useEffect(() => {
    objectPanelEventEmitter.on('GET_ACTIVE_KEY', getActiveKey);
    return () => {
      objectPanelEventEmitter.removeListener('GET_ACTIVE_KEY', getActiveKey);
    }
  }, [getActiveKey]);

  const updateObjectPanel = useCallback(() => {
    clearTimeout(updateTimeout.current);
    const time = Date.now();
    if (time - lastUpdateTime.current >= minRenderInterval) {
      lastUpdateTime.current = time;
      forceUpdate();
    } else {
      updateTimeout.current = setTimeout(() => {
        lastUpdateTime.current += minRenderInterval;
        forceUpdate();
      }, lastUpdateTime.current + minRenderInterval - time);
    }
  }, [forceUpdate]);
  useEffect(() => {
    objectPanelEventEmitter.on('UPDATE_OBJECT_PANEL', updateObjectPanel);
    return () => {
      objectPanelEventEmitter.removeListener('UPDATE_OBJECT_PANEL', updateObjectPanel);
    }
  }, [updateObjectPanel]);

  return (
    <ObjectPanelContext.Provider
      value={{
        polygonSides,
        dimensionValues: dimensionValues.current,
        activeKey,
        updateDimensionValues,
        getDimensionValues,
        updateActiveKey: setActiveKey,
        updateObjectPanel,
      }}
    >
      {children}
    </ObjectPanelContext.Provider>
  );
};
