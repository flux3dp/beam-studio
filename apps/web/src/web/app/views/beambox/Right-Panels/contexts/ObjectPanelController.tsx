import eventEmitterFactory from 'helpers/eventEmitterFactory';

const objectPanelEventEmitter = eventEmitterFactory.createEventEmitter('object-panel');

const updateDimensionValues = (newValue: any): void => {
  objectPanelEventEmitter.emit('UPDATE_DIMENSION_VALUES', newValue);
};

const getDimensionValues = (key?: string): any => {
  const response = {
    dimensionValues: {},
  };
  objectPanelEventEmitter.emit('GET_DIMENSION_VALUES', response, key);
  return response.dimensionValues;
};

const minEventInterval = 50;
let updateObjectPanelTimeout = null;
const updateObjectPanel = (): void => {
  clearTimeout(updateObjectPanelTimeout);
  updateObjectPanelTimeout = setTimeout(() => {
    objectPanelEventEmitter.emit('UPDATE_OBJECT_PANEL');
  }, minEventInterval);
};

const updatePolygonSides = (polygonSides: number): void => {
  objectPanelEventEmitter.emit('UPDATE_POLYGON_SIDES', polygonSides);
};

const updateActiveKey = (activeKey: string | null): void => {
  objectPanelEventEmitter.emit('UPDATE_ACTIVE_KEY', activeKey);
};

const getActiveKey = (): string | null => {
  const response = {
    activeKey: undefined,
  };
  objectPanelEventEmitter.emit('GET_ACTIVE_KEY', response);
  return response.activeKey;
};

export default {
  updateObjectPanel,
  updateDimensionValues,
  getDimensionValues,
  updatePolygonSides,
  updateActiveKey,
  getActiveKey,
  events: objectPanelEventEmitter,
};
