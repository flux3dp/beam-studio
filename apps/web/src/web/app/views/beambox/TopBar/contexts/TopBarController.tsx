import BeamboxGlobalInteraction from 'app/actions/beambox/beambox-global-interaction';
import eventEmitterFactory from 'helpers/eventEmitterFactory';
import { IDeviceInfo } from 'interfaces/IDevice';

const topBarEventEmitter = eventEmitterFactory.createEventEmitter('top-bar');

const setElement = (elem: Element | null): void => {
  if (!elem) {
    BeamboxGlobalInteraction.onObjectBlur();
  } else {
    BeamboxGlobalInteraction.onObjectBlur();
    BeamboxGlobalInteraction.onObjectFocus([elem]);
  }

  topBarEventEmitter.emit('SET_ELEMENT', elem);
};

const updateTitle = (title: string, isCloudFile: boolean): void => {
  topBarEventEmitter.emit('UPDATE_TITLE', title, isCloudFile);
};

const onTitleChange = (handler: (title: string, isCloudFile: boolean) => void): void => {
  topBarEventEmitter.on('UPDATE_TITLE', handler);
};

const offTitleChange = (handler: (title: string, isCloudFile: boolean) => void): void => {
  topBarEventEmitter.removeListener('UPDATE_TITLE', handler);
};

const setHasUnsavedChange = (hasUnsavedChange: boolean): void => {
  topBarEventEmitter.emit('SET_HAS_UNSAVED_CHANGE', hasUnsavedChange);
};

const getTopBarPreviewMode = (): boolean => {
  const response = {
    isPreviewMode: false,
  };
  topBarEventEmitter.emit('GET_TOP_BAR_PREVIEW_MODE', response);
  return response.isPreviewMode;
};

const getSelectedDevice = (): IDeviceInfo | null => {
  const response = {
    selectedDevice: null,
  };
  topBarEventEmitter.emit('GET_SELECTED_DEVICE', response);
  return response.selectedDevice;
};

const setSelectedDevice = (device: IDeviceInfo): void => {
  topBarEventEmitter.emit('SET_SELECTED_DEVICE', device);
};

export default {
  setElement,
  updateTitle,
  setHasUnsavedChange,
  getTopBarPreviewMode,
  getSelectedDevice,
  setSelectedDevice,
  onTitleChange,
  offTitleChange,
};
