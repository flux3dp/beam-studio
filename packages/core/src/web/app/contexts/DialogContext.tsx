import * as React from 'react';

import eventEmitterFactory from 'helpers/eventEmitterFactory';
import isWeb from 'helpers/is-web';

export const DialogContext = React.createContext({
  dialogComponents: [],
});

export const eventEmitter = eventEmitterFactory.createEventEmitter();

export class DialogContextProvider extends React.Component<any> {
  private dialogComponents: {
    id: string,
    component: JSX.Element
  }[];

  constructor(props) {
    super(props);
    this.dialogComponents = [];
  }

  componentDidMount() {
    eventEmitter.on('ADD_DIALOG_COMPONENT', this.addDialogComponent.bind(this));
    eventEmitter.on('CLEAR_ALL_DIALOG_COMPONENTS', this.clearAllDialogComponents.bind(this));
    eventEmitter.on('CHECK_ID_EXIST', this.isIdExist.bind(this));
    eventEmitter.on('POP_DIALOG_BY_ID', this.popDialogById.bind(this));
    if (isWeb()) {
      window.addEventListener('DISMISS_FLUX_LOGIN', () => {
        this.popDialogById.call(this, 'flux-id-login');
      });
    }
  }

  componentWillUnmount() {
    eventEmitter.removeAllListeners();
  }

  addDialogComponent = (id: string, dialogComponent: JSX.Element): void => {
    this.dialogComponents.push({ id, component: dialogComponent });
    this.forceUpdate();
  };

  isIdExist = (id: string, response: {
    isIdExist: boolean,
  }): void => {
    response.isIdExist = this.dialogComponents.some((dialog) => dialog.id === id);
  };

  popDialogById = (id: string): void => {
    this.dialogComponents = this.dialogComponents.filter((dialog) => dialog.id !== id);
    this.forceUpdate();
  };

  clearAllDialogComponents = (): void => {
    this.dialogComponents = [];
    this.forceUpdate();
  };

  render() {
    const { children } = this.props;
    return (
      <DialogContext.Provider
        value={{
          dialogComponents: this.dialogComponents,
        }}
      >
        {children}
      </DialogContext.Provider>
    );
  }
}
