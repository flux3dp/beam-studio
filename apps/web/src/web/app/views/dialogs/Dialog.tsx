import * as React from 'react';
import classNames from 'classnames';

import { DialogContext } from 'app/contexts/DialogContext';

const ComponentWrapper = (props) => props.children;

interface Props {
  className?: string;
}

const Dialog = ({ className = '' }: Props): JSX.Element => {
  const { dialogComponents } = React.useContext(DialogContext);
  const renderComponents = () => {
    const components = [];
    for (let i = 0; i < dialogComponents.length; i += 1) {
      const { component } = dialogComponents[i];
      components.push(
        <ComponentWrapper key={i}>
          {component}
        </ComponentWrapper>,
      );
    }
    return components;
  };

  return (
    <div className={classNames('dialog-container', className)}>
      {renderComponents()}
    </div>
  );
};

export default Dialog;
