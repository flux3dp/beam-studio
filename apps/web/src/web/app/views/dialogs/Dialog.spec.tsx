import React from 'react';
import { render } from '@testing-library/react';

import { DialogContext } from 'app/contexts/DialogContext';

import Dialog from './Dialog';

test('should render correctly', () => {
  const { container } = render(
    <DialogContext.Provider
      value={{
        dialogComponents: [
          { component: <div>Hello World</div> },
          { component: <span>Hello Flux</span> },
        ],
      }}
    >
      <Dialog className="flux" />
    </DialogContext.Provider>
  );
  expect(container).toMatchSnapshot();
});
