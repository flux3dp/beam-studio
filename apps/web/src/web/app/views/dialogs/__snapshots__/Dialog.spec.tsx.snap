// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="dialog-container flux"
  >
    <div>
      Hello World
    </div>
    <span>
      Hello Flux
    </span>
  </div>
</div>
`;
